src/tunnel.c

```diff
@@ -118,21 +118,35 @@ static int pppd_run(struct tunnel *tunnel)
 		return 1;
 	} else if (pid == 0) {
 		static const char *args[] = {
-			pppd_path, "38400", "noipdefault", "noaccomp",
-			"noauth", "default-asyncmap", "nopcomp", "receive-all",
-			"nodefaultroute", ":1.1.1.1", "nodetach",
-			"lcp-max-configure", "40", "mru", "1354",
-			NULL, NULL, NULL, NULL,
-			NULL, NULL, NULL, NULL,
-			NULL
+			pppd_path,
+			"38400", // speed
+			":1.1.1.1", // <local_IP_address>:<remote_IP_address>
+			"noipdefault",
+			"noaccomp",
+			"noauth",
+			"default-asyncmap",
+			"nopcomp",
+			"receive-all",
+			"nodefaultroute",
+			"nodetach",
+			"lcp-max-configure", "40",
+			"mru", "1354",
+			NULL, // "usepeerdns"
+			NULL, NULL, NULL, // "debug", "logfile", pppd_log
+			NULL, NULL, // "plugin", pppd_plugin
+			NULL, NULL, // "ipparam", pppd_ipparam
+			NULL // terminal null pointer required by execvp()
 		};
+
 		// Dynamically get first NULL pointer so that changes of
 		// args above don't need code changes here
 		int i = ARRAY_SIZE(args) - 1;
 		for (; args[i] == NULL; i--)
 			;
 		i++;
 
+		// Coverity detected a defect, actually a false positive:
+		//  CID 196857: Out-of-bounds write (OVERRUN)
```

Comment:
the reason is probably that coverity doesn't recognize that at least one `NULL` entry will remain in `args[]`, so that the `for` loop above will be executed at least once. 
BTW: `while(args[i] == NULL) i--;` would perhaps be easier to read than the `for` loop

Reply:
Indeed `args` is not constant and Coverity is unable to infer that `args` *will* still contain NULL elements.

Good idea about the `while`, will do.